### ISSUE TITLE
UnhandledPromiseRejection when call to SQS deleteMessageBatch rejects

### ISSUE THREAD

**Issue by jmac105** [PruvoNet/squiss-ts#50]
Your issue may already be reported!
Please search on the [issue tracker](../) before creating one.

## Expected Behavior
If the call to deleteMessageBatch rejects, then that rejected Promise should be handled properly and not bubble up.

## Current Behavior
An UnhandledPromiseRejectionWarning is logged to console, for example:
`(node:58881) UnhandledPromiseRejectionWarning: AWS.SimpleQueueService.NonExistentQueue: The specified queue does not exist for this wsdl version.`

## Possible Solution
The cause of this bug is that a rejected Promise is returned from https://github.com/PruvoNet/squiss-ts/blob/master/src/Squiss.ts#L352, however that rejected Promise is not caught at https://github.com/PruvoNet/squiss-ts/blob/master/src/Squiss.ts#L471 which is where the call to _`deleteMessages` occurs. 

Proposed solutions:
1. Return resolved promise from catch block in `_deleteMessages`, the error has already been emitted so no need to reject as well?
2. Add `.catch()` handler in `_deleteXMessages` which suppresses the error

## Steps to Reproduce (for bugs)

Can be replicated via the unit tests. For visibility set https://github.com/PruvoNet/squiss-ts/blob/master/src/test/src/index.spec.ts#L725 to be the only test to run:
`it.only('emits error when delete call fails', () => {`
Warnings can then be seen in the console
```
[me@mymachine squiss-ts (master)]$ npm test

> squiss-ts@4.0.6 test /Users/me/GitHub/squiss-ts
> npm run lint && npm run mocha


> squiss-ts@4.0.6 lint /Users/me/GitHub/squiss-ts
> tslint -c tslint.json 'src/**/*.ts' 'test/**/*.ts'


> squiss-ts@4.0.6 mocha /Users/me/GitHub/squiss-ts
> mocha --opts src/test/mocha.opts



  index
    Failures
(node:70693) UnhandledPromiseRejectionWarning: Error: test
    at Object.promise (/Users/me/GitHub/squiss-ts/src/test/src/index.spec.ts:731:41)
    at getQueueUrl.then (/Users/me/GitHub/squiss-ts/src/Squiss.ts:348:16)
    at <anonymous>
(node:70693) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 3)
(node:70693) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
      ✓ emits error when delete call fails (72ms)


  1 passing (84ms)
```

## Context
We are currently testing out squiss-ts by testing it in different scenarios before implementing in production code. A real world example which would cause an error like this would be a network blip.

## Your Environment
squiss-ts: 4.0.6
Node: 8.16.1

**Comment by regevbr** [comment]
@jmac105 thanks for reporting the issue! I will check it right away :-)

**Comment by regevbr** [comment]
@jmac105 the issue is fixed and released in 4.0.7
Thanks again for reporting!

**Comment by jmac105** [comment]
Fantastic response time, thanks @regevbr
